edweiss-app/components/courses/AssignmentDisplay.tsx

```diff
@@ -79,54 +78,21 @@ const AssignmentDisplay: ReactComponent<{ item: AssignmentWithColor, index: numb
 
     return (
         isSwipeable ?
-            <Swipeable
-                ref={(ref) => { swipeableRefs.current[index] = ref; }} renderRightActions={renderRightActions} onSwipeableOpen={(direction) => {
-                    if (direction === 'right') { console.log(`Swipe detected on assignment: ${item.name}`); saveTodo(item.name, item.dueDate, item.type); swipeableRefs.current[index]?.close(); }
-                }
-                }>
+            <Swipeable testID={testIDs.swipeableComponent}
+                ref={(ref) => { swipeableRefs.current[index] = ref; }}
+                renderRightActions={renderRightActions}
+                onSwipeableOpen={(direction) => {
```

Comment:
A logic as important as this should be extracted to its own function above for readability and refactoring (with [`useCallback`](https://react.dev/reference/react/useCallback) if you want to it to be perfect).